.github/workflows/push.yml

```
@@ -35,6 +35,15 @@ jobs:
             zts: true
     name: "LINUX_X64_${{ matrix.debug && 'DEBUG' || 'RELEASE' }}_${{ matrix.zts && 'ZTS' || 'NTS' }}"
     runs-on: ubuntu-20.04
+    env:
+      MYSQL_TEST_HOST: mysql
+      PDO_MYSQL_TEST_DSN: mysql:host=mysql;dbname=mysql
+      PDO_MYSQL_TEST_HOST: mysql
+    services:
+      mysql:
+        image: mysql:8
+        ports: [3306:3306]
+        env: { MYSQL_ROOT_PASSWORD: root }
```

As some early feedback, I dislike the inconsistent use of YAML markup. This should be equivalent to:
```suggestion
        ports:
          - 3306:3306
        env:
          MYSQL_ROOT_PASSWORD: root
```

---

I collapsed because GitHub actions requires copy/pasting this for every job. I don't care particularly, so I'll conform to your preference.